chainstate/merkle_root_providers.go

```diff
@@ -0,0 +1,65 @@
+package chainstate
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/libsv/bitcoin-hc/transports/http/endpoints/api/merkleroots"
+)
+
+// Pulse provider
+type pulseClientProvider struct {
+	url       string
+	authToken string
+}
+
+func (provider pulseClientProvider) getName() string {
+	return ProviderPulse
+}
+
+// verifyMerkleProof using Pulse
+func (provider pulseClientProvider) verifyMerkleRoots(ctx context.Context, merkleProofs []string) error {
+	return verifyWithPulse(ctx, provider, merkleProofs)
+}
+
+func verifyWithPulse(ctx context.Context, provider pulseClientProvider, merkleProofs []string) error {
+	jsonData, err := json.Marshal(merkleProofs)
+
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{}
+	req, err := http.NewRequestWithContext(ctx, "POST", provider.url, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Authorization", provider.authToken)
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close() //nolint: all // Close the body
+
+	// Parse response body.
+	var merkleRootsRes merkleroots.MerkleRootsConfirmationsResponse
+	bodyBytes, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return fmt.Errorf("error during reading response body: %s", err.Error())
+	}
+
+	err = json.Unmarshal(bodyBytes, &merkleRootsRes)
+	if err != nil {
+		return fmt.Errorf("error during unmarshaling response body: %s", err.Error())
+	}
+
+	if merkleRootsRes.AllConfirmed {
+		return nil
+	}
+	return errors.New("not all merkle roots confirmed")
```

Comment:
### suggestion, 
Worth confirming with other team members.
But maybe it would be better to limit the functionality of the client to only send request to the Pulse and return either response or error if there were any http error. And the logic that reason about the response should be made in a caller of this method.

Reply:
Done, now we check it in go-paymail

Reply:
You change it more drastically then I thought.

I thought that we will keep a PulseClient, but it's only responsibility will be to send request to Pulse and return either successful response body as a result or error if there is an HTTP error.